draft-ietf-tls-esni.md

```diff
@@ -322,6 +314,20 @@ The `ECHConfigList` structure contains one or more `ECHConfig` structures in
 decreasing order of preference. This allows a server to support multiple
 versions of ECH and multiple sets of ECH parameters.
 
+## Public Identity Validation {#identity-validate}
+
+The `public_identity` value corresponding to an ECHConfig MUST contain either
+a DNS name or IP address. Validation criteria for each of these values is below:
+
+- DNS names: The value MUST NOT begin or end with an ASCII dot and MUST be
+  parsable as a dot-separated sequence of LDH labels, as defined in
+  {{!RFC5890}}, Section 2.3.1.
+- IP addresses: The value MUST be as described in {{!RFC5280}}, Section
+  4.2.1.6.
```

Comment:
Subject Alternative Name actually leans on GeneralName, which is a pretty big sledgehammer, as discussed in #405.

Reply:
Even if we only use `iPAddress` here, it will require some carveouts.  For starters, it actually defines a subnet, rather than a single address.

I think we're better off inferring DNS-or-address from the textual representation in `public_identity`.

Reply:
I'm not leaning on GeneralName, but specifically on what's allowed for iPAddress, since that's what's relevant for certificate verification. I think the text here adequately covers both cases. I'm happy to take concrete suggestions if people have them.